scipy/linalg/tests/test_extending.py

```diff
@@ -0,0 +1,79 @@
+import os
+import platform
+import shutil
+import subprocess
+import sys
+import sysconfig
+from importlib.util import module_from_spec, spec_from_file_location
+
+import numpy as np
+import pytest
+
+from scipy.linalg.blas import cdotu
+from scipy.linalg.lapack import dgtsv
+
+try:
+    import cython
+    from Cython.Compiler.Version import version as cython_version
+except ImportError:
+    cython = None
+else:
+    from numpy._utils import _pep440
```

Comment:
It might be better to use our own version of `pep440` (`from scipy._lib import _pep440`) rather than relying on a private api